src/autocomplete.ts

```diff
@@ -43,7 +53,54 @@ export function setupAutocomplete(
             target.value.substring(end, target.value.length);
           target.selectionStart = start + auto.length;
           target.selectionEnd = start + auto.length;
+          autocompleteCaret = 0;
+        } else if (key === "]") {
+          if (candidates === null) {
+            candidates = [];
+            let matches = (s: string): boolean => {
+              return (
+                s.toLocaleLowerCase().substring(0, autoString.length) ==
+                autoString.toLocaleLowerCase()
+              );
+            };
+            for (const key in neptunesPride.universe.galaxy.stars) {
+              const star: any = neptunesPride.universe.galaxy.stars[key];
+              if (!matches(star.n)) continue;
+              candidates.push({
+                matchText: star.n,
+                completion: `[[${star.n}]]`,
+              });
+            }
+            for (const key in neptunesPride.universe.galaxy.players) {
+              const player = neptunesPride.universe.galaxy.players[key];
+              if (!matches(player.alias)) continue;
+              candidates.push({
+                matchText: player.alias,
+                completion: `[[${key}]] ${player.alias}`,
+              });
+            }
+            candidates.sort((a, b) => {
```

Comment:
Might be nice to always prefer player matches, but let's use it for a bit and see.